sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -64,59 +65,74 @@ def create_observations_with_values_in_simplex(self, experiment, on_boundary=Fal
-    best_observation_values = [dict(name="y1", value=best_value)]
-    best_observation = self.make_observation(
-      experiment=experiment,
-      assignments=best_observation_assignments,
-      values=best_observation_values,
-    )
-    observations.append(best_observation)
-    random.shuffle(observations)
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_observations = 10
+    values_list = [[dict(name="y1", value=numpy.random.rand())] for _ in range(num_observations)]
+    idx = numpy.random.randint(num_observations)
+    values_list[idx] = [dict(name="y1", value=best_value)]
+
+    for values in values_list:
+      suggestion = self.conn.experiments(e.id).suggestions().create()
+      self.conn.experiments(e.id).observations().create(
+        suggestion=suggestion.id,
+        values=values,
+      )
 
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    assert len(best_observations_from_logger) == 1
-    expected_best_observation = best_observation.get_client_observation(experiment)
-    assert expected_best_observation == best_observations_from_logger[0]
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    assert len(best_assignments_list) == 1
+    assert best_assignments_list[0].values[0].value == best_value
 
   def test_multimetric(self):
     experiment_meta = self.get_experiment_feature("multimetric")
-    experiment = LocalExperimentBuilder(experiment_meta)
-    pareto_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=True)
-    non_optimal_observations = self.create_observations_with_values_in_simplex(experiment, on_boundary=False)
-    observations = pareto_optimal_observations + non_optimal_observations
-    random.shuffle(observations)
-
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    self.assert_observations_are_sorted(best_observations_from_logger, experiment)
-    expected_pareto_optimal_observations = [o.get_client_observation(experiment) for o in pareto_optimal_observations]
-    self.assert_observation_lists_are_equal(best_observations_from_logger, expected_pareto_optimal_observations)
+    experiment_meta["type"] = "random"
+    e = self.conn.experiments().create(**experiment_meta)
+
+    num_optimal = 12
+    pareto_optimal_values = self.create_values_in_simplex(on_boundary=True, num=num_optimal)
+    non_optimal_values = self.create_values_in_simplex(on_boundary=False)
+    observation_values_list = pareto_optimal_values + non_optimal_values
+    assignments_list = [
+      self.conn.experiments(e.id).suggestions().create().assignments for _ in range(len(observation_values_list))
```

Comment:
This is not the same thing as before, since all assignments would be exactly the same.